lambda expression would cause KotlinNullPointerException

Version: 0.0.3
Code:  

based on retropiler/retropiler-demo

```
diff --git a/app/src/main/java/io/github/gfx/java8androidexample/MainActivity.java b/app/src/main/java/io/github/gfx/java8androidexample/MainActivity.java
index 46673fc..722d822 100644
--- a/app/src/main/java/io/github/gfx/java8androidexample/MainActivity.java
+++ b/app/src/main/java/io/github/gfx/java8androidexample/MainActivity.java
@@ -4,6 +4,7 @@ import android.annotation.SuppressLint;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
 import android.widget.TextView;
+import android.widget.Toast;
 
 import java.util.Optional;
 
@@ -16,6 +17,9 @@ public class MainActivity extends AppCompatActivity {
         setContentView(R.layout.activity_main);
 
         TextView textView = (TextView) findViewById(R.id.text);
+        textView.setOnClickListener(v -> {
+            Toast.makeText(this, "hoge", Toast.LENGTH_LONG).show();
+        });
 
         Optional.of("foo").map(String::toUpperCase).ifPresent(textView::setText);
     }
```


error log:

```
:app:transformClassesWithRetrolambdaForDebug (Thread[Daemon worker Thread 8,5,main]) completed. Took 0.616 secs.
:app:transformClassesWithRetropilerForDebug (Thread[Daemon worker Thread 8,5,main]) started.
:app:transformClassesWithRetropilerForDebug
Putting task artifact state for task ':app:transformClassesWithRetropilerForDebug' into context took 0.0 secs.
Task :app:transformClassesWithRetropilerForDebug class loader hash: ce342f511275207590669e436e5c81e5
Task :app:transformClassesWithRetropilerForDebug actions class loader hash: 991467d7782e1b6f326623c84a8073ee
Executing task ':app:transformClassesWithRetropilerForDebug' (up-to-date check took 0.004 secs) due to:
  Task ':app:transformClassesWithRetropilerForDebug' class path has changed from 7c76059190f16614cfff746b7bb23290 to ce342f511275207590669e436e5c81e5.
All input files are considered out-of-date for incremental task ':app:transformClassesWithRetropilerForDebug'.
:app:transformClassesWithRetropilerForDebug FAILED
:app:transformClassesWithRetropilerForDebug (Thread[Daemon worker Thread 8,5,main]) completed. Took 0.028 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithRetropilerForDebug'.
> kotlin.KotlinNullPointerException (no error message)

* Try:
Run with --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:transformClassesWithRetropilerForDebug'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:84)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:236)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:228)
        at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:61)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:228)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:215)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:77)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:58)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:32)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:113)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$3.execute(DefaultGradleLauncher.java:196)
        at org.gradle.initialization.DefaultGradleLauncher$3.execute(DefaultGradleLauncher.java:193)
        at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:193)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:119)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:102)
        at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:71)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:75)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:49)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:44)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:29)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:47)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: kotlin.KotlinNullPointerException
        at io.github.retropiler.Runtime.getRetroClass(Runtime.kt:25)
        at io.github.retropiler.Runtime.fixupLambdaClass(Runtime.kt:47)
        at io.github.retropiler.Runtime.preprocess(Runtime.kt:36)
        at io.github.retropiler.RetropilerTransform.transform(RetropilerTransform.kt:63)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:185)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:181)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
        at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:176)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$IncrementalTaskAction.doExecute(DefaultTaskClassInfoStore.java:163)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:123)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:76)
        ... 70 more


BUILD FAILED

```

--- a/plugin/src/main/kotlin/io/github/retropiler/RetropilerExprEditor.kt
+++ b/plugin/src/main/kotlin/io/github/retropiler/RetropilerExprEditor.kt
@@ -46,6 +46,8 @@ class RetropilerExprEditor(val classPool: ClassPool) : ExprEditor() {
 
     override fun edit(m: MethodCall) {
         val method = m.method
+        val declaringClass = method.declaringClass
+        val retroClass = runtime.getRetroClassOrNull(declaringClass) ?: return
 
         trace(m)
 
@@ -56,9 +58,6 @@ class RetropilerExprEditor(val classPool: ClassPool) : ExprEditor() {
             """)
             return
         }
-
-        val declaringClass = method.declaringClass
-        val retroClass = runtime.getRetroClassOrNull(declaringClass) ?: return
 
         val signature = makeRetroSignature(m.signature)
 

--- a/plugin/src/main/kotlin/io/github/retropiler/Runtime.kt
+++ b/plugin/src/main/kotlin/io/github/retropiler/Runtime.kt
@@ -60,7 +60,7 @@ class Runtime(val classPool: ClassPool) {
     // retrolambda generates `Consumer lambdaFactory$(...)` so it replaces the return type
     // to retropiler runtime classes
     private fun fixupLambdaClass(lambdaClass: CtClass) {
-        val retroClass = getRetroClass(lambdaClass.interfaces[0])
+        val retroClass = getRetroClassOrNull(lambdaClass.interfaces[0]) ?: return
 
         lambdaClass.addInterface(retroClass)
 
@@ -153,8 +153,7 @@ class Runtime(val classPool: ClassPool) {
             return
         }
 
-        // assume that lambdaClass implements just the single functional interface
-        val retroClass = getRetroClass(ctClass.interfaces[0])
+        val retroClass = getRetroClassOrNull(ctClass.interfaces[0]) ?: return
 
         ctClass.interfaces = ctClass.interfaces.filter {
             it == retroClass

--- a/test/src/main/java/io/github/retropiler/test/MainActivity.java
+++ b/test/src/main/java/io/github/retropiler/test/MainActivity.java
@@ -47,4 +47,10 @@ public class MainActivity extends AppCompatActivity {
         Optional<String> optStr = Optional.of("baz");
         optStr.ifPresent(s -> text.append("Optional#ifPresent: " + s));
     }
+
+    void f() {
+        // just places it here for the regression: https://github.com/retropiler/retropiler/pull/10
+        Runnable task = () -> {};
+        task.run();
+    }
 }

